docs/self-hosted-getting-started.md

```diff
@@ -0,0 +1,99 @@
+          OPENAI_API_KEY: your-api-key
+        depends_on:
+          - db
+    ```
+
+2. Connect to your db instance with your DB client of choice. We need to enable the pgai extension and create a simple blog table with the following schema:
+    ```sql
+    CREATE EXTENSION IF NOT EXISTS ai CASCADE;
+    CREATE TABLE blog (
+        id SERIAL PRIMARY KEY,
+        title TEXT,
+        authors TEXT,
+        contents TEXT,
+        metadata JSONB
+    );
+    ```
+   
+3. Insert some data into your new table. E.g. you can run these insert statements:
+    ```sql
+    INSERT INTO blog (title, authors, contents, metadata)
+    VALUES
+    ('Getting Started with PostgreSQL', 'John Doe', 'PostgreSQL is a powerful, open source object-relational database system...', '{"tags": ["database", "postgresql", "beginner"], "read_time": 5, "published_date": "2024-03-15"}'),
+
+    ('10 Tips for Effective Blogging', 'Jane Smith, Mike Johnson', 'Blogging can be a great way to share your thoughts and expertise...', '{"tags": ["blogging", "writing", "tips"], "read_time": 8, "published_date": "2024-03-20"}'),
+
+    ('The Future of Artificial Intelligence', 'Dr. Alan Turing', 'As we look towards the future, artificial intelligence continues to evolve...', '{"tags": ["AI", "technology", "future"], "read_time": 12, "published_date": "2024-04-01"}'),
+
+    ('Healthy Eating Habits for Busy Professionals', 'Samantha Lee', 'Maintaining a healthy diet can be challenging for busy professionals...', '{"tags": ["health", "nutrition", "lifestyle"], "read_time": 6, "published_date": "2024-04-05"}'),
+
+    ('Introduction to Cloud Computing', 'Chris Anderson', 'Cloud computing has revolutionized the way businesses operate...', '{"tags": ["cloud", "technology", "business"], "read_time": 10, "published_date": "2024-04-10"}'); 
+    ```
+
+4. Now we need to create a vectorizer for the table. The vectorizer worker will pick this one up and create the corresponding embeddings for us:
+    ```sql
+    SELECT ai.create_vectorizer(
+       'blog'::regclass,
+       destination => 'blog_contents_embeddings',
+       embedding => ai.embedding_openai('text-embedding-3-small', 768),
+       chunking => ai.chunking_recursive_character_text_splitter('contents'),
+       scheduling => ai.scheduling_none()
+    );
+    ```
+    If you check the logs of the vectorizer worker, you should see that it has picked up the table and is processing it.
+
+
+5. You'll need to restart the vectorizer now, since it fails to run if pgai is not installed or no vectorizer is defined yet. A simple `docker-compose up -d` should do the trick.
+
+
+6. Now we can run a simple semantic search query to see the embeddings in action:
+    ```sql
+    SELECT
+        chunk,
+        embedding <=>  ai.openai_embed('text-embedding-3-small', 'pgai', _dimensions=>768) as distance
+    FROM blog_contents_embeddings
+    ORDER BY distance;
+    ```
```

Comment:
There is a bit of a problem where this query will return empty results if the worker hasn't run yet. I think I will add to the compose that the cli runs every 5 seconds for this tutorial.
Right now this is not an issue because of the restart above which instantly triggers another embedding run.

Reply:
We can tell them to run:

```
select ai.vectorizer_queue_pending(1) \watch
```

To monitor how many pending items are left. When the query returns `0` then they can proceed.

Reply:
Also added this in the troubleshooting section 👍